tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs

```diff
@@ -1,37 +1,486 @@
+                        GitHandler.Run(config, $"add -A .");
+                        GitHandler.Run(config, $"commit -m \"Automatic asset update from test-proxy.\"");
+                        GitHandler.Run(config, $"push origin {config.AssetsRepoBranch}");
+                    }
+                    catch (GitProcessException e)
+                    {
+                        throw GenerateInvokeException(e.Result);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Restores a set of recordings from the assets repo. Honors configuration of assets.json passed into it.
+        /// </summary>
+        /// <param name="pathToAssetsJson"></param>
+        /// <returns></returns>
+        public async Task Restore(string pathToAssetsJson) {
+            var config = await ParseConfigurationFile(pathToAssetsJson);
+            var initialized = config.IsAssetsRepoInitialized();
+
+            if (!initialized)
+            {
+                InitializeAssetsRepo(config);
+            }
+
+            CheckoutRepoAtConfig(config);
+        }
+
+        /// <summary>
+        /// Resets a cloned assets repository to the default contained within the assets.json targeted commit.
+        /// </summary>
+        /// <param name="pathToAssetsJson"></param>
+        /// <returns></returns>
+        public async Task Reset(string pathToAssetsJson) {
+            var config = await ParseConfigurationFile(pathToAssetsJson);
+            var initialized = config.IsAssetsRepoInitialized();
+            var allowReset = true;
+
+            if (!initialized)
+            {
+                InitializeAssetsRepo(config);
+            }
+
+            var pendingChanges = DetectPendingChanges(config);
+
+            if (pendingChanges.Length > 0)
+            {
+                // TODO: Azure/azure-sdk-tools/3698
+            }
+
+            if (allowReset)
+            {
+                try
+                {
+                    GitHandler.Run(config, "checkout *");
+                    GitHandler.Run(config, "git clean -xdf");
+                }
+                catch(GitProcessException e)
+                {
+                    throw GenerateInvokeException(e.Result);
+                }
+
+                if (!string.IsNullOrWhiteSpace(config.SHA))
+                {
+                    CheckoutRepoAtConfig(config);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Given a CommandResult, generate an HttpException.
+        /// </summary>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        public HttpException GenerateInvokeException(CommandResult result)
+        {
+            var message = $"Invocation of \"git {result.Arguments}\" had a non-zero exit code {result.ExitCode}.\nStdOut: {result.StdOut}\nStdErr: {result.StdErr}\n";
+
+            return new HttpException(HttpStatusCode.BadRequest, message);
```

Comment:
I wonder if it's better to do the http exception conversions at the API controller layer when you're calling the store methods? Otherwise every store implementation has to return the same exceptions. Also, 400 vs 500?